File: gestionart_backend_ruben_y_jessica/api/src/main/java/com/gestionart/api/domain/models/Notificacion.java
package com.gestionart.api.domain.models;

import java.time.LocalDateTime;

import com.gestionart.api.domain.enums.TipoNotificacion;

public class Notificacion {
    private Long id;
    private Long vendedorid;
    private TipoNotificacion tipo; 
    private boolean leido;
    private LocalDateTime fecha;
    
    public Notificacion() {
    }

    public Notificacion(Long id, Long vendedorid, TipoNotificacion tipo, boolean leido, LocalDateTime fecha) {
        this.id = id;
        this.vendedorid = vendedorid;
        this.tipo = tipo;
        this.leido = leido;
        this.fecha = fecha;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getVendedorId() {
        return vendedorid;
    }

    public void setVendedorId(Long vendedorid) {
        this.vendedorid = vendedorid;
    }

    public TipoNotificacion getTipo() {
        return tipo;
    }

    public void setTipo(TipoNotificacion tipo) {
        this.tipo = tipo;
    }

    public boolean isLeido() {
        return leido;
    }

    public void setLeido(boolean leido) {
        this.leido = leido;
    }

    public LocalDateTime getFecha() {
        return fecha;
    }

    public void setFecha(LocalDateTime fecha) {
        this.fecha = fecha;
    }

    public Long getVendedorid() {
        return vendedorid;
    }

    public void setVendedorid(Long vendedorid) {
        this.vendedorid = vendedorid;
    }

    
}

File: gestionart_frontend_ruben_y_jessica/lib/data/models/Notificacion.dart
import 'package:gestionart_frontend_ruben_y_jessica/data/enums/TipoNotificacion.dart';

class Notificacion {
  final int id;
  final int vendedorId;
  final Tiponotificacion tipo;
  final bool leido;
  final DateTime fecha;

  Notificacion({
    required this.id, 
    required this.vendedorId, 
    required this.tipo, 
    required this.leido, 
    required this.fecha
  });
  
  int getId() {
    return id;
  }
  
  int getVendedorId() {
    return vendedorId;
  }
  
  Tiponotificacion getTipo() {
    return tipo;
  }
  
  bool isLeido() {
    return leido;
  }
  
  DateTime getFecha() {
    return fecha;
  }

  factory Notificacion.fromJson(Map<String, dynamic> json) {
    return Notificacion(
      id: json['id'] ?? 0,
      vendedorId: json['vendedorId'] ?? 0,
      tipo: Tiponotificacion.values.firstWhere(
        (e) => e.toString().split('.').last == (json['tipo'] ?? 'NUEVO_PEDIDO'),
        orElse: () => Tiponotificacion.NUEVO_PEDIDO,
      ),
      leido: json['leido'] ?? false,
      fecha: json['fecha'] != null 
          ? DateTime.parse(json['fecha']) 
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'vendedorId': vendedorId,
      'tipo': tipo.toString().split('.').last,
      'leido': leido,
      'fecha': fecha.toIso8601String(),
    };
  }
}